models/models.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const db  = require("../server/db");
+const Promise = require("bluebird");
+
+class Communication {
+  static findById(id) {
+    return new Promise((fulfill, reject) => {
+      db("comms")
+      .where("commid", id)
+      .limit(1)
+      .then(function (comms) {
+        if (comms.length > 0) {
+          fulfill(comms[0])
+        } else {
+          fulfill()
+        }
+      })
+      .catch(reject)      
+    })
+  }
+}
+
+class Message {
+  static create(messageObject) {
+    return new Promise((fulfill, reject) => {
+      db("msgs")
+      .insert(messageObject).returning("msgid")
+      .then((messageIds) => {
+        fulfill(messageIds[0])
+      }).catch(reject)
+    })
+  }
+}
+
+class Convo {
+  static closeAll(cmid, clid) {
+    return new Promise((fulfill, reject) => {
+      db("convos")
+        .where("client", clid)
+        .andWhere("cm", cmid)
+        .andWhere("convos.open", true)
+        .pluck("convid")
+        .then(function (convos) {
+          db("convos").whereIn("convid", convos)
+            .update({
+              open: false
+            }).then(function (success) {
+              fulfill(success)
+            })
+            .catch(reject)
+        })
+        .catch(reject)
```

Comment:
Sticking with the routes pattern, we can probably return the future on line 45 and then have a single catch statement.